lang/LangPrimSource/PyrListPrim.cpp

```diff
@@ -44,6 +44,8 @@ int class_identdict_index, class_identdict_maxsubclassindex;
 PyrClass *class_identdict;
 PyrSymbol *s_proto, *s_parent;
 PyrSymbol *s_delta, *s_dur, *s_stretch;
+PyrSymbol *s_type, *s_rest, *s_empty, *s_r;
+PyrClass *class_rest, *class_metarest;
```

Comment:
Here are my raw benchmarks. The first column uses an event with five items in addition to the item from the left header; the second column has 100 items + the one other. For the raw results, I ran each test five times and took the median.

```
** Updated primitive with ~\type == \rest~ and external getsym's
|               |                 5 |               100 |
|---------------+-------------------+-------------------|
| dur = Rest(1) | 0.010426011999996 | 0.015488212000001 |
| \type, \rest  | 0.010418496999989 | 0.010615183000006 |
| degree = \r   |       0.011263352 | 0.034184259999996 |
| dur = 1       |       0.012290433 | 0.037585183000004 |

** Updated primitive, ~\type == \rest~ in primitive
|               |                 5 |               100 |
|---------------+-------------------+-------------------|
| dur = Rest(1) | 0.027805868999963 | 0.030072043000018 |
| \type, \rest  |  0.01601790199993 |        0.01555836 |
| degree = \r   | 0.028085984000086 | 0.050382741000021 |
| dur = 1       | 0.038235088999954 | 0.063535976999901 |
```

```
(
a = [
	[
		[0.010426011999996, 0.015488212000001],
		[0.010418496999989, 0.010615183000006],
		[0.011263352, 0.034184259999996],
		[0.012290433, 0.037585183000004],
	], [
		[0.027805868999963, 0.030072043000018],
		[0.01601790199993, 0.01555836],
		[0.028085984000086, 0.050382741000021],
		[0.038235088999954, 0.063535976999901],
	]
];

b = a[0].collect { |row, i| a[1][i] / row };

"Speed ratios per test".postln;
b.do(_.postln);

c = b.flat;
"\nMinimum speed ratio = %, maximum = %\n".postf(c.minItem, c.maxItem);

"Average improvements: %\n".postf(b.mean);
""
)

->
Speed ratios per test
[ 2.6669707458589, 1.9416084309807 ]
[ 1.5374484438539, 1.4656704458125 ]
[ 2.4935724285351, 1.4738578807915 ]
[ 3.1109635437542, 1.690452777625 ]

Minimum speed ratio = 1.4656704458125, maximum = 3.1109635437542
Average improvements: [ 2.4522387905005, 1.6428973838024 ]
```

We must be running different benchmarks, then. I'm seeing anywhere from 1.4x to 3x faster.

I have no problem with making them `static` variables inside the function. (I didn't know you could do that.)